api/routes/auth.py

```diff
@@ -121,16 +121,16 @@ def apple_auth(auth_data: s.AppleAuthTokenIn, db: Session = Depends(get_db)):
     decoded_token = c.verify_apple_token(auth_data)
 
     user = db.scalar(
-        sa.select(m.User).where(
-            m.User.email == decoded_token.email,
+        sa.select(m.AuthAccount).where(
+            m.AuthAccount.email == decoded_token.email,
```

Comment:
you should also filter AuthAcount by type and search only for type apple and oauth_id to match from decoded token